sdk/azidentity/azidentity.go

```diff
@@ -156,12 +175,17 @@ func newDefaultMSIPipeline(o ManagedIdentityCredentialOptions) azcore.Pipeline {
 	if o.HTTPClient == nil {
 		o.HTTPClient = azcore.DefaultHTTPClientTransport()
 	}
-
+	var statusCodes []int
 	// retry policy for MSI is not end-user configurable
 	retryOpts := azcore.RetryOptions{
 		MaxTries:   5,
 		RetryDelay: 2 * time.Second,
-		StatusCodes: append(azcore.StatusCodesForRetry[:],
+		StatusCodes: append(statusCodes,
```

Comment:
Why did these codes get duplicated?

Reply:
So, during the last review, we discussed possibly removing the 503 status code for retry since it would probably consume more time than we would (ideally?) want. I decided to copy the values over instead of doing an operation over the slice, because I think it'll be safer this way and this code doesnt need to check the other slice in case something is added before or after the 503 status code.

Reply:
Ah I didn't realize this is a subset, makes sense that it's copied.  Might be worth adding a comment indicating it's a subset of `StatusCodesForRetry`.

Reply:
Added